Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -170,27 +170,24 @@ private void RunWhileSuspended(IEnumerable<TestMethod> tests)
                             continue;
                         }
 
-                        var stopwatch = new Stopwatch();
-                        stopwatch.Start();
-
                         try
                         {
                             fakes.StartTest();
                             RunInternal(testInitialize);
-                            test.Run();
+                            var result = test.Run();
+                            // we can trigger this event, because cleanup can fail without affecting the result
+                            OnTestCompleted(test, result);
```

Comment:
This makes sense from the implementation POV but from consumer's POV, it seems confusing since if something went wrong in the cleanup, it might be indicative of a flawed code, rather than flawed test? I'd want to know that a test is running to completion but can't be cleaned up so I can see if I need to fix the code under the test.

Reply:
yea, the problem with failed cleanup is the possibility of trouble down the line in other unit-tests... cleanup usually implies that there is some shared state between unit-tests, which is ... acceptable, but can result in such issues.

At that point it might be useful to actually modify a success-result to include  a failure message?

Reply:
My vote would be to output this as a separate line. That way when I click on the line in the toolwindow, it'll take  me to the broken `@TestCleanup` for further diagnosis. If we modify the test's result, I would be taken to the test itself, which isn't where the ~princess~ problem is.